backend/app/core/auth.py

```diff
@@ -8,16 +8,18 @@
 from app.db import session
 from app.db.crud.user import get_user_by_email
 from app.db.models import User
+from app.db.schemas.user import JWTUser
```

Comment:
The main change here is that there are now 2 ways to inject a user dependency in the API endpoints:
- get_current_active_user 
- get_current_active_db_user
(and their superuser analogues)

The former is quick (only a JWT decode) and the latter is heavy (fetches user from DB).
This sped up the doc upload quite a bit, as it doesn't need the DB user, but the request is kept open for a long time as Boto uploads PDFs to S3. This starved the DB pool with the original dependency.